chsdi/views/loader.py

```diff
@@ -16,7 +16,13 @@ def loadjs(request):
     ignore_polyfill = request.params.get('ignore_polyfill')
     # Determined automatically in subscriber
     lang = request.lang
-    loader_js_bucket_host = request.registry.settings['loader_js_bucket_host']
+
+    if "loader_js_bucket_localhost" in request.registry.settings:
+        host = request.registry.settings["loader_js_bucket_localhost"]
+    elif "CloudFront-Forwarded-Proto" in request.headers:
+        host = '%s://%s' % (request.headers["CloudFront-Forwarded-Proto"], request.host)
```

Comment:
You should also add an `elif "X-Forwarded-Proto" in request.headers:` which is the standard header for such use case.

Also rename the `host` variable to `base_url`. `host` usually refer to an hostname and not a full url with protocol.

Reply:
But it would be much better to solve this in a generic way like in our Flask application, where the header check is done for all request object. Otherwise we need to implement this check everytime we need to generates url !

Reply:
In our Flask we have the header check set in gunicorn, maybe  we can do this in Apache as well, because we wants to get rid of apache I would only check for Cloudfront-Forwarded-Proto in Apache, later we moving to gunicorn we can use the same logic as in our Flask application: https://github.com/geoadmin/service-shortlink/blob/a8786e675a9c01584a2473fe05c92bf48251a78c/wsgi.py#L55

Reply:
> You should also add an `elif "X-Forwarded-Proto" in request.headers:` which is the standard header for such use case.
> 
> Also rename the `host` variable to `base_url`. `host` usually refer to an hostname and not a full url with protocol.

How about `host_url`, which would be Pyramid-like: https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/webob.html#urls

Reply:
simply using `host_url` won't solve it. It is correct to use such variable in code, but for know we have the issue that CF send Cloudfront-Forwarded-Proto instead of X-Forwarded-Proto and that Apache don't set the correct scheme in wsgi.url_scheme !

Reply:
Yes, as discussed, the generic way might be best. I just meant as a variable name. Using `host_url` should be quite the way, it is used by Pyramid, too.